src/inmanta/execute/proxy.py

```diff
@@ -69,12 +115,31 @@ class DynamicProxy:
     by native code.
     """
 
-    def __init__(self, instance: "Instance") -> None:
+    def __init__(self, instance: "Instance", *, context: Optional[ProxyContext] = None) -> None:
+        """
+        :param instance: The object to proxy.
+        :param context: The context this object lives in.
+        """
         object.__setattr__(self, "__instance", instance)
+        object.__setattr__(self, "__context", context if context is not None else ProxyContext())
```

Comment:
should be cache the default `ProxyContext()`? we may have a LOT of it?

Reply:
With the paths there's not much to cache. Memory impact of the proxy contexts seems to be negligible. Construction overhead may be a few percent, to be further investigated.